frontend/components/site/CommitSummary.jsx

```diff
@@ -0,0 +1,79 @@
+import React, { useEffect } from 'react';
+import { useSelector, connect } from 'react-redux';
+
+import PropTypes, { BUILD } from 'prop-types';
+import { IconBranch } from '../icons';
+import LoadingIndicator from '../LoadingIndicator';
+
+import buildActions from '../../actions/buildActions';
+import { timeFrom, dateAndTime } from '../../util/datetime';
+
+function buildShaLink(owner, repo, sha) {
+  const BASE = 'https://github.com';
+  const baseHref = `${BASE}/${owner}/${repo}`;
+  const linkHref = `${baseHref}/commit/${sha}`;
+  return (
+    <a
+      className="sha-link"
+      href={linkHref}
+      title={sha}
+      target="_blank"
+      rel="noopener noreferrer"
+    >
+      { sha.slice(0, 7) }
+    </a>
+  );
+}
+
+function CommitSummary({ buildId }) {
+  const { isLoading, data: buildDetails } = useSelector(state => state.build);
+
+  useEffect(() => {
+    buildActions.fetchBuild(buildId);
+  }, [buildId]);
+
+  if (isLoading) {
+    return (<LoadingIndicator size="mini" text="Getting commit details..." />);
+  }
+
+  return (
+    (!isLoading && buildDetails && (
+    <div className="commit-summary">
+      <h3 className="commit-branch">
+        <IconBranch />
+        {' '}
+        {buildDetails.branch}
+      </h3>
+      <p className="commit-details">
+        {buildShaLink(
+          buildDetails.site.owner,
+          buildDetails.site.repository,
+          buildDetails.clonedCommitSha
+        )}
+        &nbsp;by&nbsp;
+        <b className="commit-username">{buildDetails.username}</b>
+&nbsp;
+        <span className="commit-time" title={dateAndTime(buildDetails.createdAt)}>
+          { timeFrom(buildDetails.createdAt) }
+        </span>
+      </p>
+    </div>
+    ))
+  );
+}
+CommitSummary.propTypes = {
+  buildId: PropTypes.number.isRequired,
+  build: PropTypes.shape({
+    isLoading: PropTypes.bool,
+    data: PropTypes.objectOf(BUILD),
+  }),
+};
+
+CommitSummary.defaultProps = {
+  build: null,
+};
+
+const mapStateToProps = ({ build }) => ({ build });
+
+export { CommitSummary };
+export default connect(mapStateToProps)(CommitSummary);
```

Comment:
For future consideration. We need to decide/document how we connect global state to components. Should we keep everything in `frontend/components` as pure components and pass all their props down, should we restructure them into their own directories (ie `frontend/containers`), only connect to the state at the page level, or leave it as is?

Reply:
Yeah, I really wondered what the 'right' path for this was, too. It seemed like we connected state and props for better unit testing, but @drewbo has been working on a better way to mount the components for testing, iirc...

Reply:
I think the Redux-era convention was global state, one reducer for each "smart component" which roughly existed at a page-level, and then pure components below that. It seems like React Hooks are moving folks away from even persisting a global state but don't necessarily specify at which level state should be maintained. The current [React docs about state](https://react.dev/learn/managing-state) aren't super prescriptive about this and at times seem to reproduce a lot of Redux ideas, so I'd favor a longer conversation before we look into a bigger refactor here.

Reply:
Started tracking the creation of a future standard in https://github.com/cloud-gov/pages-core/issues/4332